### DISCUSSION TITLE
CI/CD to Azure tenant, fetch client error

### DISCUSSION THREAD

**Discussion by tabit01** [SharePoint/sp-dev-docs#8292] [Q&A]
I wish to implement a Sharepoint Framework CI/CD flow for our Sharepoint Framework solutions using Azure DevOps Pipelines. I am following [MS CI/CD docs](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/toolchain/implement-ci-cd-with-azure-pipelines) and [PnP CLI login docs](https://pnp.github.io/cli-microsoft365/user-guide/connecting-office-365/)

Our goal is to create a deployment flow without interaction or using credentials. Here is a snippet of the deploy template yaml:
 
```
- script: |
    export CLIMICROSOFT365_AADAPPID=<AAD APP ID>
    export CLIMICROSOFT365_TENANT=<TENANT ID>
    printenv
  displayName: Set environment variables
- script: m365 login --authType secret --secret <SECRET VALUE>
  displayName: Login to Microsoft 365
```

The deployment fails in this task, presenting this error:

```
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/ce9d1fdf-66ef-4bb5-bb9a-7dfbaef33d52.sh
Error: network_error: Network request failed. Please check network trace to determine root cause. | Fetch client threw: Error: HTTP status code 401 | Attempted to reach: https://login.microsoftonline.com/common/oauth2/v2.0/token
##[error]Bash exited with code '1'.
Finishing: Login to Microsoft 365
```
Does anyone have related experience, or useful debugging directions?

**Comment by nanddeepn** [comment]
Hi @tor-axbit 
The script sample example [here](https://pnp.github.io/script-samples/spo-install-deploy-spfx-solution/README.html?tabs=cli-m365-ps) might help you.
